### PR TITLE
Automated Live test for Call Automation

### PR DESCRIPTION
**Pull request by minwoolee-msft**
Automated Live tests for call automation
- Implemented infrastructure for call automation's automated live test
- Uses ServiceBus and dispatcher for message broker
- See the wiki page (shared internally) for how to run this locally

### CODE REVIEW

**Comment by williamzhao87** [comment]
FYI @minwoolee-msft, I added some recording unmixed audio live tests to this PR:
image

File: sdk/communication/Azure.Communication.CallingServer/tests/CallRecording/CallRecordingAutomatedLiveTests.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                Assert.AreEqual(CallConnectionState.Connected, properties.Value.CallConnectionState);
+
+                // try start recording unmixed audio - no channel affinity
+                var startRecordingResponse = await client.GetCallRecording().StartRecordingAsync(
+                    new StartRecordingOptions(new ServerCallLocator(properties.Value.ServerCallId))
+                    {
+                        RecordingChannel = RecordingChannel.Unmixed,
+                        RecordingContent = RecordingContent.Audio,
+                        RecordingFormat = RecordingFormat.Wav,
+                        // RecordingStateCallbackEndpoint = "DISPATCHER CALLBACK URI?" TODO: Add Dispatcher's callback URI here, since this is OutOfCall API
```
Comment (williamzhao87) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallRecording/CallRecordingAutomatedLiveTests.cs:78]: FYI, will we need to add support OutOfCall APIs for recording on the dispatcher somehow?

File: sdk/communication/Azure.Communication.CallingServer/tests/CallRecording/CallRecordingAutomatedLiveTests.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                    });
+                Assert.AreEqual(StatusCodes.Status200OK, startRecordingResponse.GetRawResponse().Status);
+                Assert.NotNull(startRecordingResponse.Value.RecordingId);
+
+                // try stop recording
+                var stopRecordingResponse = await client.GetCallRecording().StopRecordingAsync(startRecordingResponse.Value.RecordingId);
+                Assert.AreEqual(StatusCodes.Status204NoContent, stopRecordingResponse.Status);
+
+                // wait for CallRecordingStateChanged event TODO: Dispatcher should receive this callback event
```
Comment (williamzhao87) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallRecording/CallRecordingAutomatedLiveTests.cs:87]: Not sure if Dispatcher needs to be updated to receive the CallRecordingStateChanged event

File: sdk/communication/Azure.Communication.CallingServer/tests/CallAutomationClient/CallAutomationClientAutomatedLiveTests.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        public async Task CreateCallToACSGetCallAndHangUpCallTest()
+        {
+            /* Test case: ACS to ACS call
+             * 1. create a CallAutomationClient.
+             * 2. create a call from source to one ACS target.
+             * 3. get updated call properties and check for the connected state.
+             * 4. hang up the call.
+             * 5. once call is hung up, verify disconnected event
+            */
+            if (SkipCallingServerInteractionLiveTests)
```
Comment (yyw-msft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallAutomationClient/CallAutomationClientAutomatedLiveTests.cs:32]: nit: change SkipCallingServerInteractionLiveTests to SkipCallAutomationInteractionLiveTests

File: sdk/communication/Azure.Communication.CallingServer/tests/CallAutomationClient/CallAutomationClientAutomatedLiveTests.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                Assert.Ignore("Skip callingserver interaction live tests flag is on.");
```
Comment (yyw-msft) [inline sdk/communication/Azure.Communication.CallingServer/tests/CallAutomationClient/CallAutomationClientAutomatedLiveTests.cs:33]: nit: change callingserver -> callautomation

**Comment by yyw-msft** [comment]
FYI this was done while u were away but after @juntuchen-msft added the CallAutomation project in SDK, we should add the automated live tests there

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallAutomationClientAutomatedLiveTestsBase.cs [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                }
+            }
+
+            // complete the message. messages is deleted from the queue.
+            await args.CompleteMessageAsync(args.Message);
+        }
+
+        private Task ErrorHandler(ProcessErrorEventArgs args)
+        {
+            return Task.CompletedTask;
```
Comment (yyw-msft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallAutomationClientAutomatedLiveTestsBase.cs:203]: do we just silently ignore errors? should we at least log in console?

File: sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallAutomationClientAutomatedLiveTestsBase.cs [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        /// </summary>
+        /// <returns>The instrumented <see cref="CallAutomationClientOptions" />.</returns>
+        private CallAutomationClientOptions CreateServerCallingClientOptionsWithCorrelationVectorLogs()
+        {
+            CallAutomationClientOptions callClientOptions = new CallAutomationClientOptions();
+            callClientOptions.Diagnostics.LoggedHeaderNames.Add("MS-CV");
+            return InstrumentClientOptions(callClientOptions);
+        }
+
+        private string ParseFromAndTo(string unparsed)
```
Comment (yyw-msft) [inline sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallAutomationClientAutomatedLiveTestsBase.cs:217]: nit: is this the Unique Id? would it be more clear to name the function ParseUniqueId?

**Pull request closed without merging**